src/sidebar/services/frame-sync.js

```diff
@@ -50,63 +50,67 @@ export default function FrameSync($rootScope, $window, store, bridge) {
    * notify connected frames about new/updated/deleted annotations.
    */
   function setupSyncToFrame() {
-    let prevPublicAnns = 0;
+    let prevPublicAnnotationCount = 0;
 
     watch(
       store.subscribe,
-      [() => store.getState().annotations.annotations, () => store.frames()],
+      [
+        () => store.getState().annotations.annotations,
+        () => store.frames(),
+        () => store.frameFetchStatus(),
+      ],
       ([annotations, frames], [prevAnnotations]) => {
```

Comment:
This callback should get triggered by changes to `store.frameFetchStatus()` returned value, but we don't need access to the value itself here; note two-element arg array `[annotations, frames]`—is this kosher?

Reply:
Its kosher in the sense that it is is perfectly legal JS ;)

However, "fake" dependencies may be a sign that some refactoring is required. In this context we have one `watch` "effect" that does two different things: 1) Load or unload annotations from the page, 2) Notify the page about how many public annotations there are in total.

I sketched out what it might look like if the two "effects" were separated into separate `watch` calls which track only their respective dependencies (nb. not tested):

```js
/**
   * Watch for changes to the set of annotations displayed in the sidebar and
   * notify connected frames about new/updated/deleted annotations.
   */
  function setupSyncToFrame() {
    watch(
      store.subscribe,
      () => store.getState().annotations.annotations,
      (annotations, prevAnnotations) => {
        const sidebarAnnotations = new Set();
        const addedAnnotations = [];

        annotations.forEach(annotation => {
          if (metadata.isReply(annotation)) {
            // The frame does not need to know about replies
            return;
          }

          sidebarAnnotations.add(annotation.$tag);
          if (!inFrame.has(annotation.$tag)) {
            addedAnnotations.push(annotation);
          }
        });

        const deletedAnnotations = prevAnnotations.filter(
          annotation => !sidebarAnnotations.has(annotation.$tag)
        );

        // We currently only handle adding and removing annotations from the frame
        // when they are added or removed in the sidebar, but not re-anchoring
        // annotations if their selectors are updated.
        if (addedAnnotations.length) {
          bridge.call('loadAnnotations', addedAnnotations.map(formatAnnot));
          addedAnnotations.forEach(annotation => inFrame.add(annotation.$tag));
        }

        deletedAnnotations.forEach(annotation => {
          bridge.call('deleteAnnotation', formatAnnot(annotation));
          inFrame.delete(annotation.$tag);
        });
      }
    );

    const publicAnnotationCount = createSelector(
      annotations =>
        annotations.filter(a => !metadata.isReply(a) && metadata.isPublic(a))
          .length
    );

    /** Have annotations been fetched for all frames? */
    const isFetchComplete = createSelector(fetchStatus =>
      Object.values(fetchStatus).every(status => status)
    );

    watch(
      store.subscribe,
      [
        () => publicAnnotationCount(store.getState().annotations.annotations),
        () => isFetchComplete(store.frameFetchStatus()),
      ],
      ([count, fetchComplete]) => {
        if (!fetchComplete) {
          return;
        }
        bridge.call(bridgeEvents.PUBLIC_ANNOTATION_COUNT_CHANGED, count);
      }
    );
  }
```

Reply:
Yeah, that seems fine; I am all for decoupling things that don't need to be glommed.